src/machine/machinectl.c

```diff
@@ -254,19 +274,43 @@ static int list_machines(int argc, char *argv[], void *userdata) {
 
         qsort_safe(machines, n_machines, sizeof(MachineInfo), compare_machine_info);
 
+        /* Allocate for prefix max characters for all fields + spaces between them + strlen(",\n") */
+        prefix = new0(char, max_name + max_class + max_service + max_os + max_version_id + 5 + strlen(",\n"));
+        if (!prefix) {
+                r = log_oom();
+                goto out;
+        }
+        r = sprintf(prefix, "%-*s",
+                        (int) (max_name +
+                        max_class +
+                        max_service +
+                        max_os +
+                        max_version_id + 5 + strlen(",\n")),
+                        ",\n");
```

Comment:
instead of manually allocating the buffer, and then using `sprintf()` on it, please use `asprintf()` right-away which combines these two steps in one.